lib/matplotlib/widgets.py

```diff
@@ -347,13 +349,22 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt='%1.2f',
             self.poly = ax.axvspan(valmin, valinit, 0, 1, **kwargs)
             self.vline = ax.axvline(valinit, 0, 1, color='r', lw=1)
 
-        self.valfmt = valfmt
-        ax.set_yticks([])
         if orientation == 'vertical':
             ax.set_ylim((valmin, valmax))
+            axis = ax.yaxis
         else:
             ax.set_xlim((valmin, valmax))
+            axis = ax.xaxis
+
+        self.valfmt = valfmt
+        self._fmt = axis.get_major_formatter()
```

Comment:
Is this attribute is introduced here? If so, let's call it
```suggestion
        self._formatter = axis.get_major_formatter()
```

Reply:
`fmt` for `format`(ter) is a pretty standard abbreviation, and this avoids an awkward linebreak a bit later which also helps from a legibility PoV.  OTOH I don't really want to argue too much over this so if you are sure you prefer _formatter, let me know and I'll do the change (yes, it's a new attribute, so we can call it however we want).

Reply:
If I see `fmt` I expand to `format` in my mind, but I do not expand to `formatter`.   OTOH, this isn't really a big deal here.